cypress/support/PersistentLogs.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+ */
+
+var testCaseRegistry = [];
+var indexRegistry = 0;
+var logEntries = '';
+
+// Override the original console.log() function
+var originalConsolLog = console.log;
+
+// This becomes the effective console.log() function.
+console.log = function (message) {
+  originalConsolLog.apply(console, arguments);
+  logEntries += message + '\r\n';
+};
+
+// This will be called by the test list manager to register each test case
+// in the order the tests will be run in.
+Cypress.PersistentLogs = {};
```

Comment:
I noticed this pattern in the other reviews. Why is it needed?

It seems it's only there to share functions defined here with the test list manager. Is that true or does it serve some other purpose?
If so it seems similar to writing global variables on the browser `window` object and being able to reference them anywhere.  Similarly with concept of over writing of browser defaults or using global variables it's generally avoided.

Are there any alternatives to sharing the function?

Reply:
Well the normal the alternative to global variables is to actually define a module and `import` or `require` it so it's an explicit dependency.  I think in a previous review you said this doesn't work but I couldn't remember. It may not be an option but I wanted ask because we need to try to limit this pattern.  It seems like we found global variables is working for what you needed and the more we add them we just contribute to the tech debt we eventually want to get away from.

Reply:
The `import`/`require` model you suggest does not work in this case. 

I believe I have used this technique sparingly, only when necessary, but as I continue to work with the code I will keep it in mind and remove any instances of it that are not necessary.